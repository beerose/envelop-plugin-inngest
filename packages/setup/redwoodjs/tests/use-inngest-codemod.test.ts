const path = require('path');
const fs = require('fs');
const jscodeshift = require('jscodeshift');
const transform = require('../src/use-inngest-codemod');

describe('When using use-inngest-codemod to transform, the graphQL handler', () => {
  const defaultJscodeshiftOpts = {
    verbose: 1,
    dry: false,
    print: true,
    babel: false,
    extensions: 'js',
    ignorePattern: '**/node_modules/**',
    ignoreConfig: [],
    runInBand: false,
    silent: false,
    parser: 'ts',
    parserConfig: {},
    failOnError: true,
    stdin: false,
  };

  describe('transforms code correctly', () => {
    it('when given a standard default GraphQL handler', () => {
      const source = fs.readFileSync(path.join(__dirname, '__testfixtures__', 'default.input.js'), 'utf8');
      const output = transform(
        { path: '../__testfixtures__/default.input.js', source },
        { jscodeshift },
        { ...defaultJscodeshiftOpts }
      );
      expect(output).toMatchSnapshot();
    });

    it('when GraphQL handler already configures some extraPlugins', () => {
      const source = fs.readFileSync(path.join(__dirname, '__testfixtures__', 'extra-plugins.input.js'), 'utf8');
      const output = transform(
        { path: '../__testfixtures__/extra-plugins.input.js', source },
        { jscodeshift },
        { ...defaultJscodeshiftOpts }
      );

      expect(output).toMatchSnapshot();
    });

    it('when GraphQL handler already has useInngest setup', () => {
      const source = fs.readFileSync(path.join(__dirname, '__testfixtures__', 'exists.input.js'), 'utf8');
      const output = transform(
        { path: '../__testfixtures__/exists.input.js', source },
        { jscodeshift },
        { ...defaultJscodeshiftOpts }
      );
      expect(output).toMatchSnapshot();
    });
  });
});
