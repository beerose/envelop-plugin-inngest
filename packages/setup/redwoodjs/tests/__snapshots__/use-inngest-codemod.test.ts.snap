// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When using use-inngest-codemod to transform, the graphQL handler transforms code correctly when GraphQL handler already configures some extraPlugins 1`] = `
"import { createGraphQLHandler } from '@redwoodjs/graphql-server';

import directives from 'src/directives/**/*.{js,ts}';
import sdls from 'src/graphql/**/*.sdl.{js,ts}';
import services from 'src/services/**/*.{js,ts}';

import { getCurrentUser } from 'src/lib/auth';
import { db } from 'src/lib/db';
import { logger } from 'src/lib/logger';

import { foo } from 'src/lib/foo';

import { inngestPlugin } from \\"src/inngest/plugin\\";

export const handler = createGraphQLHandler({
  getCurrentUser,
  loggerConfig: { logger, options: {} },
  directives,
  sdls,
  services,
  extraPlugins: [foo, inngestPlugin],

  onException: () => {
    // Disconnect from your database with an unhandled exception.
    db.$disconnect();
  }
});
"
`;

exports[`When using use-inngest-codemod to transform, the graphQL handler transforms code correctly when GraphQL handler already has useInngest setup 1`] = `
"import { createGraphQLHandler } from \\"@redwoodjs/graphql-server\\";

import directives from \\"src/directives/**/*.{js,ts}\\";
import sdls from \\"src/graphql/**/*.sdl.{js,ts}\\";
import services from \\"src/services/**/*.{js,ts}\\";

import { getCurrentUser } from \\"src/lib/auth\\";
import { db } from \\"src/lib/db\\";
import { logger } from \\"src/lib/logger\\";

import { inngestPlugin } from \\"src/inngest/plugin\\";

export const handler = createGraphQLHandler({
  getCurrentUser,
  loggerConfig: { logger, options: {} },
  directives,
  sdls,
  services,
  extraPlugins: [inngestPlugin],

  onException: () => {
    // Disconnect from your database with an unhandled exception.
    db.$disconnect();
  }
});
"
`;

exports[`When using use-inngest-codemod to transform, the graphQL handler transforms code correctly when given a standard default GraphQL handler 1`] = `
"import { createGraphQLHandler } from '@redwoodjs/graphql-server'

import directives from 'src/directives/**/*.{js,ts}'
import sdls from 'src/graphql/**/*.sdl.{js,ts}'
import services from 'src/services/**/*.{js,ts}'

import { getCurrentUser } from 'src/lib/auth'
import { db } from 'src/lib/db'
import { logger } from 'src/lib/logger'

import { inngestPlugin } from \\"src/inngest/plugin\\";

export const handler = createGraphQLHandler({
  getCurrentUser,
  loggerConfig: { logger, options: {} },
  directives,
  sdls,
  services,

  onException: () => {
    // Disconnect from your database with an unhandled exception.
    db.$disconnect()
  },

  extraPlugins: [inngestPlugin]
})
"
`;
